R/try_regional_data.R

```diff
@@ -0,0 +1,125 @@
+#' uk <- try_regional_data("United Kingdom", class = TRUE)
+#' uk
+#'
+#' # return UK data from the class object]
+#' uk$return()
+#' }
+try_regional_data <- function(country, level = "1", totals = FALSE,
+                              localise = TRUE, steps = FALSE,
+                              class = FALSE, verbose = TRUE, regions,
+                              include_level_2_regions = deprecated(),
+                              localise_regions = deprecated(),
+                              ...) {
+  if (is_present(include_level_2_regions)) {
+    deprecate_warn(
+      "0.9.0",
+      "covidregionaldata::try_regional_data(include_level_2_regions = )", "covidregionaldata::try_regional_data(level = )"
+    )
+    if (include_level_2_regions) {
+      level <- "1"
+    } else {
+      level <- "2"
+    }
+  }
+
+  if (is_present(localise_regions)) {
+    deprecate_warn(
+      "0.9.0",
+      "covidregionaldata::try_regional_data(localise_regions = )", "covidregionaldata::try_regional_data(localise = )"
+    )
+    localise <- localise_regions
+  }
+
+  # construct short hand options
+  title_country <- str_to_title(country)
+  nospace <- str_replace_all(title_country, " ", "")
+  targets <- c(
+    title_country, toupper(title_country), nospace, toupper(nospace)
+  )
+  
+  # check we have data for desired class
+  datasets <- covidregionaldata::get_available_datasets("regional")
+  target_class <- bind_rows(
+    filter(datasets, map_lgl(.data$class, ~ any(str_detect(., targets)))),
+    filter(datasets, map_lgl(.data$origin, ~ any(str_detect(., targets))))
+  ) %>%
+    distinct()
+
+  if (nrow(target_class) != 0) {
+    # check data availability and initiate country class if available
+    region_class <- initialise_dataclass(
+      class = country, level = level, regions = regions,
+      totals = totals, localise = localise,
+      verbose = verbose, steps = steps, get = TRUE,
+      type = "regional", ...
+    )
+
+    return(
+      return_data(region_class, class = class)
+    )
+  } else {
+    google_sources <- csv_reader(Google$public_fields$common_data_urls$index)
+
+    google_inputs <-
+      filter(google_sources, map_lgl(.data$country_name, ~ any(str_detect(., targets))))
+    if (nrow(google_inputs) != 0) {
+      message_verbose(verbose = verbose,
+                      "No country-specific data class found. Getting data from Google for ",
+                      country)
+      google_class <- initialise_dataclass(
+        class = "Google", level = sprintf("%d", as.integer(level)+1),
+        totals = totals, localise = TRUE,
+        verbose = verbose, steps = steps,
+        regions = country, get = TRUE,
+        type = "national", ...
+      )
+      return_data(google_class, class = class)
+    } else {
+      stop(
+        "No data available for ", country, ", including from Google.\n",
+        "See get_available_datasets(type = c(\"regional\")) for supported datasets."
```

Comment:
add suggestion for how to check Google for support (perhaps custom function checking Google/JHU above could be exposed to the use and called something like `check_metasources`?)